examples/simple_system/ibex_simple_system.cc

```diff
@@ -17,6 +17,7 @@ int main(int argc, char **argv) {
   simctrl.SetTop(&top, &top.IO_CLK, &top.IO_RST_N,
                  VerilatorSimCtrlFlags::ResetPolarityNegative);
 
+  svSetScope(svGetScopeFromName("TOP.ibex_simple_system"));
```

Comment:
Can you move that into the `ibex_pcount_string()` routine? (Ideally, restore the previous scope there as well after the function is done to reduce side effects, but that's probably overkill and could impact performance. Maybe add a comment to the `ibex_pcount_string() doxygen comment instead.)

Reply:
Ah, I wanted to comment there why I didn't do it, but forgot: I think that this would reduce the flexibility of usage in systems where the toplevel is not `ibex_simple_system`. Would you suggest to:

1. have an init function pcount_init (or even better, put it into a class with constructor), or
2. get the toplevel string from `verilated_top.h`?

The reason I put it here was my thinking that we maybe want to use the root scope as default (contrary to no scope). This was the third option that I chose to reduce the number of changes and still come up with a reasonable solution, forgot to mention there.

What do you prefer?

Reply:
I see, let's not go overboard then. Can you move the `svSetScope()` function bit further down to the reading of the perf counters, and add a comment why it is needed? That should match what we have today roughly, so good enough.

Reply:
Option 1 and 2 would allow to not hardcode the toplevel name into the generic files obviously.

Reply:
(We do the "full generic" solution in `VerilatorMemUtil`, but I wouldn't put that much effort into reading performance counters from a test system.)

Reply:
yeah, right, that's what I thought too